dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -827,9 +829,67 @@ def validate_job(self, job_id):
 
 def update_tas_ids(model_class, submission_id):
     sess = GlobalDB.db().session
+
     submission = sess.query(Submission).filter_by(submission_id=submission_id).one()
+    start_date = submission.reporting_start_date
+    end_date = submission.reporting_end_date
+    day_after_end = end_date + timedelta(days=1)
+
+    logger.info({
+        'message': 'Setting up TAS links',
+        'message_type': 'ValidatorInfo',
+        'submission_id': submission_id,
+        'model_class': str(model_class)
+    })
+    start = time.time()
+
+    update_query = """
+        WITH relevant_tas AS  (
+            SELECT
+                min(tas_lookup.account_num) AS min_account_num,
+                allocation_transfer_agency,
+                agency_identifier,
+                beginning_period_of_availa,
+                ending_period_of_availabil,
+                availability_type_code,
+                main_account_code,
+                sub_account_code
+            FROM
+                tas_lookup
+            WHERE
+                (('{start}'::date, '{end}'::date) OVERLAPS
+                    (tas_lookup.internal_start_date, coalesce(tas_lookup.internal_end_date, '{day_after_end}'::date)))
+            GROUP BY
+                allocation_transfer_agency,
+                agency_identifier,
+                beginning_period_of_availa,
+                ending_period_of_availabil,
+                availability_type_code,
+                main_account_code,
+                sub_account_code
+        )
+        UPDATE {model}
+        SET tas_id = min_account_num
```

Comment:
Is this update run on data which already contains valid `tas_id` values?

Reply:
These are only run on new values we just added that don't have `tas_id` values

Reply:
good to know, thanks